main/src/com/google/refine/importing/ImportingUtilities.java

```diff
@@ -327,7 +329,11 @@ public String handleResponse(final ClassicHttpResponse response) throws IOExcept
                         };
 
                         HttpClient httpClient = new HttpClient();
-                        if (httpClient.getResponse(urlString, null, responseHandler) != null) {
+                        Header acceptHeader = new BasicHeader(HttpHeaders.ACCEPT, parameters.get("accept"));
```

Comment:
You shouldn't be assuming a pre-determined set of headers here. You need to rely either on what's been submitted from the UI or the supported headers currently registered (from HttpHeadersSupport)

Reply:
This is still only using three specific headers (accept, user-agent and authorization). However, OpenRefine isn't limited to these three headers. Instead OpenRefine should support any headers that are registered via HttpHeadersSupport.

Therefor this could should either be relying on what's being submitted from the UI or work from the supported headers currently registered. So to do this you need to find all the headers that have been submitted/are supported here and then check if there is a value for them.

You can see how this works in https://github.com/OpenRefine/OpenRefine/blob/577caabec1cce968df56156d7561a1ffde4b01c4/main/src/com/google/refine/operations/column/ColumnAdditionByFetchingURLsOperation.java#L142

Reply:
Please correct me if I am wrong, I think this is still the old commits, I push other commits as well, whether I have updated this - https://github.com/OpenRefine/OpenRefine/pull/3911/files

Reply:
@sjathin - ah - thank you. Can see the latest changes address this. Will re-review when I get a chance